client/src/components/ViewContact.jsx

```diff
@@ -86,36 +66,30 @@ const ViewContact = ({ selectedContactId, setContacts }) => {
         }
       );
 
+      if (!response.ok) {
+        throw new Error("Failed to edit contact");
+      }
+
       await response.json();
 
       const refreshedListData = await fetchContacts();
       setContacts(refreshedListData);
-
-      if (!response.ok) {
-        throw new Error("Failed to edit contact");
-      }
```

Comment:
Notice how I moved this code up above `await response.json()` - if that fetch code has an error, handle it as soon as possible (don't continue on to uselessly call `await fetchContacts()` if you know the first fetch code has an error).